ssl/statem/statem_srvr.c

```diff
@@ -355,6 +372,29 @@ static int send_server_key_exchange(SSL_CONNECTION *s)
     return 0;
 }
 
+/*
+ * Should we send a CompressedCertificate message?
+ *
+ * Valid return values are:
+ *   1: Yes
+ *   0: No
+ */
+static int get_compressed_certificate_alg(SSL_CONNECTION *sc)
+{
+#ifndef OPENSSL_NO_COMP_ALG    
+    int *alg = sc->ext.compress_certificate_from_peer;
+
+    if (sc->s3.tmp.cert == NULL)
+        return TLSEXT_comp_cert_none;
+
+    for (; *alg != TLSEXT_comp_cert_none; alg++) {
+        if (sc->s3.tmp.cert->comp_cert[*alg] != NULL)
```

Comment:
So we find the most preferred alg from the peer for which we have a compressed cert and send that? i.e. peer preference order? Is that in the docs somewhere?

Reply:
I will double-check (and clarify), but yes, peer preference. Preferences are used to indicate what you can/want to receive, so it makes sense to send what the peer prefers. Given that server certificates are pre-compressed, it makes little difference to the server what it prefers, as long as it's available.

Reply:
There is something there, but I elaborated on it.

Reply:
It is interesting that RFC8879 makes no reference to the order of the algorithms in the list implying some kind of preference. I assumed this was how it worked until I checked, and it doesn't actually say anything AFAICT.

An alternative strategy could be to use the shortest compressed certificate that the peer supports. I don't mind either way though.

Reply:
The client order is used, mostly because it may be the algorithm which they have the fastest implementation of, regardless of the data size.